### ISSUE TITLE
non final methods check

### ISSUE THREAD

**Issue by marandaneto** [JakeWharton/nopen#9]
hey @JakeWharton first of all thanks for the great work.

I'd love to see _nopen_ also checking non-final methods when they should be.
I believe it makes sense to be within this lib, but this is you to say.

The idea is pretty much similar to @Open, but for methods.
A method should be final (when possible) or annotated.

I've made a Draft PR (more like a POC), #8 and happy to chat about.
If you are willing to bring this in and accept a PR, I'd gladly work on it.

Thanks a lot :)

**Comment by JakeWharton** [comment]
So this would require that any `public` or `protected` function in an `@Open` class was annotated or marked `final`?

**Comment by marandaneto** [comment]
yes, I'd like to double-check on compile-time that I'm not making an accidental overriding.
on the same line as Item 19 of Effective Java, I believe a method should explicitly be marked as overridable/open or be final.
This is a very handy check on compile-time when you ship a public API as a standalone library (binary).
